app/main.py

```diff
@@ -1,2 +1,88 @@
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Any, Union
+
+
+@dataclass
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.nodes = [[], [], [], [], [], [], [], []]
+
+    def resize(self) -> None:
+        temp = deepcopy(self.nodes)
+        size = len(self.nodes)
+        self.nodes = []
+        for _ in range(2 * size):
+            self.nodes.append([])
+        for node in temp:
+            for item in node:
+                if item:
+                    self.__write(item[0], item[1])
+
+    def __setitem__(self,
+                    key: Union[int, float, bool, str, tuple],
+                    value: Any) -> None:
+        if self.nodes.count([]) <= 1 / 3 * len(self.nodes):
+            self.resize()
+        self.__write(key, value)
+
+    def __write(self,
+                key: Union[int, float, bool, str, tuple],
+                value: Any) -> None:
+        write = True
+        for item in self.nodes[hash(key) % len(self.nodes)]:
+            if key in item:
+                item[1] = value
+                write = False
+                break
+        if write:
+            self.nodes[hash(key) % len(self.nodes)].append([key, value])
+
+    def __getitem__(self, key: Union[int, float, bool, str, tuple]) -> Any:
+        for node in self.nodes[hash(key) % len(self.nodes)]:
+            if key in node:
+                return node[1]
+        raise KeyError
+
+    def __len__(self) -> int:
+        result = 0
+        for node in self.nodes:
+            for item in node:
+                if item:
+                    result += 1
+        return result
```

Comment:
This loop takes O(n^2) time for every check.
it returns not the correct result
You should optimize it to work as O(1) :(

Reply:
It always returns correct result. But i agree it takes O(nˆ2) time

Reply:
O(1) done